pkgs/development/python-modules/mkdocs-macros-plugin/default.nix

```diff
@@ -0,0 +1,76 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  setuptools,
+  pytestCheckHook,
+  hjson,
+  jinja2,
+  mkdocs,
+  packaging,
+  pathspec,
+  python-dateutil,
+  pyyaml,
+  termcolor,
+  super-collections,
+  mkdocs-test,
+}:
+
+buildPythonPackage rec {
+  pname = "mkdocs-macros-plugin";
+  version = "1.3.9";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "fralau";
+    repo = "mkdocs-macros-plugin";
+    tag = "v${version}";
+    hash = "sha256-bL7oWWDoF+zH34XSwFY2H9op/97zO43HS+oO6lNFEr4=";
+  };
+
+  # we do not have bew and mkdocs-d2-plugin is also not packaged in nixpkgs,
+  # so we skip the assertion
+  postPatch = ''
+    substituteInPlace test/plugin_d2/test_t2.py \
+      --replace-fail 'brew' 'false'
+  '';
+
+  build-system = [
+    setuptools
+  ];
+
+  dependencies = [
+    hjson
+    jinja2
+    mkdocs
+    packaging
+    pathspec
+    python-dateutil
+    pyyaml
+    termcolor
+    super-collections
+  ];
+
+  pythonImportsCheck = [
+    "mkdocs_macros"
+  ];
+
+  nativeCheckInputs = [
+    pytestCheckHook
+    mkdocs-test
+  ];
+
+  disabledTests = [
+    # was not able to get them running
+    "test_pages"
+    "test_strict"
```

Comment:
Add at least `mkdocs-material` and `mkdocs-macros-test` to `nativeCheckInputs`.
You can see the error message using `pytestFlags = [ "-vvv" ]`

Reply:
Thanks, I managed to get all tests working except: `test_pages`.

I've verified that this test works in docker with a venv. I will look into that in the next week or so.